jobs/resources-search-index/index.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { ResourcesJobProcessorError } from '@tech-matters/hrm-job-errors';
+import { indexDocument } from '@tech-matters/elasticsearch-client';
+
+// eslint-disable-next-line prettier/prettier
+import type { SQSBatchResponse, SQSEvent, SQSRecord } from 'aws-lambda';
+
+export const handler = async (event: SQSEvent): Promise<SQSBatchResponse> => {
+  const response: SQSBatchResponse = { batchItemFailures: [] };
+
+  const processRecord = async (sqsRecord: SQSRecord) => {
+    const message = JSON.parse(sqsRecord.body);
+
+    const { accountSid, document } = message;
+    try {
+      await indexDocument({
+        indexType: 'resources',
+        id: document.id,
+        document,
+        accountSid,
+      });
+    } catch (err) {
+      console.error(new ResourcesJobProcessorError('Failed to process record'), err);
+
+      response.batchItemFailures.push({
+        itemIdentifier: sqsRecord.messageId,
+      });
+    }
+  };
+
+  const promises = event.Records.map(async sqsRecord => processRecord(sqsRecord));
+  await Promise.all(promises);
```

Comment:
Is there a small change that 2 changes to the same record could be processed out of order if we process the whole batch in parallel? 

I suppose the requests will be dispatched in order so the chances of them going out of order over the wire are pretty low & the impact of an issue is probably not so high

Reply:
FIFO queues won't make messages with the same group ID visible to a consumer until messages before them have been successful processes. We'll likely just set group id to the resource I'd, so once we add in the FIFO bits no single batch should include records for a single resource.

Reply:
I went ahead and threw untested PoC of fifo setup into a followup at #342